### PR TITLE
CLN: clean up adfvalues, avoid using `eval`

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: statsmodels/tsa/adfvalues.py [hunk 1]
```diff
@@ -223,127 +273,138 @@ def mackinnonp(teststat, regression="c", N=1, lags=None):
...
+     [-5.84215, -30.8119, -14.938, 316.006]],
+    [[-6.83743, -55.2861, -152.651, 792.577],  # N = 11
+     [-6.31396, -40.5507, -46.771, 487.185],
+     [-6.03921, -33.8950, -9.122, 285.164]],
+    [[-7.02582, -59.6037, -166.368, 989.879],  # N = 12
+     [-6.50353, -44.0797, -47.242, 543.889],
+     [-6.22941, -36.9673, -10.868, 418.414]]]
 tau_ctt_2010 = asarray(tau_ctt_2010)
 
+_tau_2010s = {
```
Comment (bashtage) [inline statsmodels/tsa/adfvalues.py]: Why reason why this is _? The rest of the tau variable's aren't, and there is an __all__ to help.

Comment (jbrockmendel) [reply statsmodels/tsa/adfvalues.py]: Accident.  I lean towards privatizing the other, am OK with making this one public.  Any preference?

**Comment by bashtage** [comment]
Might as well leave public.

**Comment by jbrockmendel** [comment]
Comments addressed and green.

LMK if I can help with anything release-specific.  ATM my presumption is pure-cleanups are among the few things safe to do immediately before a release, but fortunately we’re running out of those (after 5692 we’ll have all the W codes)

**Pull request merged**